zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -314,7 +320,44 @@ where
 
             let miner_fee = miner_fee(&mempool_txs);
 
+            // Calling state with `ChainInfo` request for relevant chain data
+            let request = ReadRequest::ChainInfo;
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                                data: None,
+                            })?;
+
+            let chain_info = match response {
+                ReadResponse::ChainInfo(Some(chain_info)) => chain_info,
+                _ => unreachable!("lets hope for always some until later"),
+            };
+
+            // Get the tip data from the state call
+            let tip_height = chain_info.tip.0;
+            let tip_hash = chain_info.tip.1;
+
             let block_height = (tip_height + 1).expect("tip is far below Height::MAX");
+            let min_time = chain_info.median_time_past.checked_add_signed(Duration::seconds(1))
+                .expect("median time plus a small constant is far below i64::MAX");
+
+            // > For each block at block height 2 or greater on Mainnet, or block height 653606 or greater on Testnet, nTime
+            // > MUST be less than or equal to the median-time-past of that block plus 90 * 60 seconds.
+            //
+            // We ignore the height as we are checkpointing on nu5 in Mainnet and Testnet.
```

Comment:
The mandatory checkpoint is Canopy, and the default checkpoint is past NU5. They are both a long way above block 653606, so this is just a documentation fix.

```suggestion
            // We ignore the height as we are checkpointing on Canopy or higher in Mainnet and Testnet.
```